compiler/luci/pass/src/SubstituteSqueezeToReshapePass.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/SubstituteSqueezeToReshapePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+/**
+ * @brief Return TRUE if all dim is known && dim > 0 && at least dim 1 exist
+ */
+bool can_squeeze_shape(luci::CircleNode *node)
+{
+  int dim_1_cnt = 0;
+  for (uint32_t r = 0; r < node->rank(); ++r)
+  {
+    if (not node->dim(r).known())
+      return false;
+    if (node->dim(r).value() < 1)
```

Comment:
If `node` shape is [1, 0, 2, 1, 3], `Squeeze` result is [0, 2, 3] which can be squeezed.

Reply:
Um? you mean we should also enable of dim value 0 ?

Reply:
I wanted to know why zero dimension is treated as special case.
If this condition is used, following example cannot be optimized.

```python
x = tf.constant(1, shape=[1, 0, 2, 1, 3])
y = tf.squeeze(x)
print(y) # tf.Tensor([], shape=(0, 2, 3), dtype=int32)
```

Reply:
ah... just wanted to substitute for normal shapes :)
as I know 0 is used for unknown value so I'm not sure replace with Reshape is OK or not for this case...

Reply:
Ah, 0 in `x = tf.constant(1, shape=[1, 0, 2, 1, 3])` is not unknown dimension but real zero dimension :D
Unknown dimension is represented as `None` such as `p = tf.compat.v1.placeholder(shape=[None, 1], dtype=tf.int32)`.

- zero dimension : `dim.known() == true && dim.value() == 0`
- unknown dimension : `dim.known() == false` (dim.value() is 0 but meaningless)

Reply:
so you mean we only need to check it's known or not...OK :)